### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by IKarbowiak**
Drop deprecated code that was marked for being removed in 2.10.

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for d2076a2ca965dba5b3de40f770a6476a29fe651b (mirumee/saleor @ remove-deprecated-code)
Base comparison is b0a0a04b4e0ab228f354b7d0b8b9023dfa096787.



<details><summary>**Found 2 differences!** (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	         50	         50	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
+ product details                            	         15	         14	              2
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
- retrieve variant list                      	         15	         16	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: tests/api/test_product.py [hunk 1]
```diff
@@ -517,7 +479,7 @@ def test_filter_product_by_category(user_api_client, product):
     category = product.category
     query = """
     query getProducts($categoryId: ID) {
-        products(categories: [$categoryId], first: 1) {
+        products(filter: {categories: [$categoryId]}, first: 1) {
```
Comment (fowczarek) [inline tests/api/test_product.py]: Probably duplicated test with `test_products_query_with_filter_category`

File: tests/api/test_product.py [hunk 2]
```diff
@@ -588,10 +550,10 @@ def test_fetch_unpublished_product_anonymous_user(api_client, unavailable_produc
 def test_filter_products_by_attributes(user_api_client, product):
```
Comment (fowczarek) [inline tests/api/test_product.py]: Probably duplicated test with `test_products_query_with_filter_attributes`

File: tests/api/test_product.py [hunk 3]
```diff
@@ -644,7 +608,7 @@ def test_filter_products_by_categories(user_api_client, categories_tree, product
     product.save()
     query = """
     query {
-        products(categories: ["%(category_id)s"], first: 1) {
+        products(filter: {categories: ["%(category_id)s"]}, first: 1) {
```
Comment (fowczarek) [inline tests/api/test_product.py]: Probably duplicated test with `test_products_query_with_filter_category`

File: tests/api/test_product.py [hunk 4]
```diff
@@ -665,7 +629,7 @@ def test_filter_products_by_collections(user_api_client, collection, product):
     collection.products.add(product)
     query = """
     query {
-        products(collections: ["%(collection_id)s"], first: 1) {
+        products(filter: {collections: ["%(collection_id)s"]}, first: 1) {
```
Comment (fowczarek) [inline tests/api/test_product.py]: Probably duplicated test with `test_products_query_with_filter_collection`

**Pull request merged**